### PR TITLE
PR for #160: Change genres to a many-to-many

### PR DESCRIPTION
**Pull request by vmoros**
### Description

* Changed the genres field of the Concert model to a many-to-many
* Updated the pull_seatgeek_data command to work with this new many-to-many field
* Deleted the genres.txt file because it has been replaced by the Genre model
* Added the Concert model to the admin interface

### Testing Directions
```
python manage.py makemigrations
python manage.py migrate
python manage.py pull_seatgeek_data
python manage.py make_users
```

1.  See that the pull_seatgeek_data command does not give any errors.
1. Look at the Concert model in the admin interface and see that the genres are no longer a string. It now has a multi-select widget because it's now a many-to-many under the hood.
> image

### Miscellaneous Comments

* These changes broke the genres filter on the events page. That's expected. I'll fix that on #153.

### Checklist

- [x] I have written tests which have maintained/increased test coverage.
- [x] I ran `git merge develop` before submitting this PR.
- [x] I understand that I may have to run `git merge develop` again after submitting this PR since new changes may have been pulled into `develop`.
- [x] I understand that it is MY responsibility to make sure that this PR does not have any conflicts.

### CODE REVIEW

File: showup/management/commands/pull_seatgeek_data.py [hunk 1]
```diff
@@ -107,14 +87,8 @@ def handle(self, *args, **options):
...
-        genres = sorted(existing_genres)
-
-        # Add genres to Genre model.
-
-        for genre in genres:
-            Genre(genre=genre).save()
-
-        # Add genres to genres.txt.
-        Command.WriteGenre(genres)
+                    # we have to create the Concert object before adding its genres
```
Comment (mksavic) [inline showup/management/commands/pull_seatgeek_data.py]: where are you making a Concert object here?

File: showup/models.py [hunk 2]
```diff
@@ -2,6 +2,16 @@
 from django.contrib.auth.models import AbstractUser
 
 
+class Genre(models.Model):
+    genre = models.TextField()
+
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=["genre"], name="Genre_unique")]
```
Comment (mksavic) [inline showup/models.py]: what does this code do

File: showup/models.py [hunk 3]
```diff
@@ -26,21 +36,14 @@ def __str__(self):
...
-
 class CustomUser(AbstractUser):
     date_of_birth = models.DateField(null=True, blank=True)
     gender = models.CharField(max_length=255, blank=True)
     interested = models.ManyToManyField(Concert, related_name="interested", blank=True)
     going = models.ManyToManyField(Concert, related_name="going", blank=True)
     bio = models.TextField(max_length=500, default="", blank=True)
     swipes = models.ManyToManyField("self", through="Swipe", symmetrical=False)
-    genres = models.ManyToManyField(Genre, related_name="genres", blank=True)
+    genres = models.ManyToManyField(Genre, related_name="fav_genres", blank=True)
```
Comment (mksavic) [inline showup/models.py]: did it need to be renamed?

File: showup/management/commands/pull_seatgeek_data.py (hunk 1 above)
Comment (vmoros) [reply showup/management/commands/pull_seatgeek_data.py]: Lines 69-77 and 85

File: showup/models.py (hunk 2 above)
Comment (vmoros) [reply showup/models.py]: It's a unique constraint so that we won't have any genre more than once

File: showup/models.py (hunk 3 above)
Comment (vmoros) [reply showup/models.py]: Yes because otherwise there would be a conflict with the related name of the genres field in the Concert model

**Comment by vmoros** [comment]
@mksavic, if we reset the DB, we will lose all genres and they will not come back until we pull another event with that genre or add the genre manually in the admin interface.

Regarding the migrations, I think you're right, I messed it up. I just fixed it though.

File: showup/models.py (hunk 3 above)
Comment (mksavic) [reply showup/models.py]: ok thx

File: showup/models.py (hunk 2 above)
Comment (mksavic) [reply showup/models.py]: ok thx

File: showup/management/commands/pull_seatgeek_data.py (hunk 1 above)
Comment (mksavic) [reply showup/management/commands/pull_seatgeek_data.py]: ok thx

**Comment by Vedanth15** [comment]
LGTM

**Pull request merged**